### PR TITLE
Sign in redirect

### PR DESCRIPTION
**Pull request by yuw006**
in case of 500 (email is associated with account) or 403 (already signed in) following would occur
task

### CODE REVIEW

File: frontend/src/pages/EventSignInPage/components/EventSignInForm/index.js [hunk 1]
```diff
@@ -10,6 +10,10 @@ import styles from './styles';
 
 import { MajorDropdownField, AffiliateDropdownField } from '@SharedComponents';
 
+import { ButtonWithConfirmationModal } from '@SharedComponents';
```
Comment (thaigillespie) [inline frontend/src/pages/EventSignInPage/components/EventSignInForm/index.js:13]: `<ButtonWithConfirmationModal` comes from `@SharedComponents` so you can put it in the curly braces on line 11 as well

File: frontend/src/pages/EventSignInPage/components/EventSignInForm/index.js [hunk 2]
```diff
@@ -10,6 +10,10 @@ import styles from './styles';
...
+import * as ROUTES from '@Constants/routes';
```
Comment (thaigillespie) [inline frontend/src/pages/EventSignInPage/components/EventSignInForm/index.js:14]: Please group path alias imports into 1 separate group, and library imports into another separate group i.e. put react-router import statement after line 6 and put `@Constants/routes` import statement after line 11

File: frontend/src/pages/EventSignInPage/components/EventSignInForm/index.js [hunk 3]
```diff
@@ -22,19 +26,58 @@ const INITIAL_INPUT_VALUES = {
 const EventSignInForm = props => {
   const { classes, handleSubmit } = props;
 
+  const [modalDisplay, setModalDisplay] = useState(false);
+
+  const history = useHistory();
```
Comment (thaigillespie) [inline frontend/src/pages/EventSignInPage/components/EventSignInForm/index.js:31]: No need for spaces between short const initialization like this. Group the const statements on lines 27, 29 and 31 together and put them at the beginning of the function body, with the hooks going before the destructuring statement (useState first, then useHistory)

File: frontend/src/pages/EventSignInPage/index.tsx [hunk 4]
```diff
@@ -58,15 +58,15 @@ function EventSignInPage(): JSX.Element {
 
           <Grid item>
             <EventSignInForm
-              handleSubmit={(values: AppUserEventRequest) =>
-                signInToEvent(eventID, values)
-              }
-            />
+              handleSubmit={(values: AppUserEventRequest, callback) =>
+                signInToEvent(eventID, values).then().catch((error) => {
+                  callback((error.status === 403) || (error.status === 500));
```
Comment (thaigillespie) [inline frontend/src/pages/EventSignInPage/index.tsx:63]: What did you do to get status code 500 for the response? Looking at the endpoint handler and stuff in the backend, it should be returning 404 and stuff, so I'm just curious as to how you're getting 500s

**Review by thaigillespie — CHANGES_REQUESTED** [review]
Looks pretty good so far. Left some comments for you to look at

**Review by audreygao — APPROVED** [review]
looks good to me

**Comment by audreygao** [comment]
@thaigillespie just wondering if we have a document for general rules and coding styles like these so that we're more aware of the styles while we code and don't have to change it every time in PR. I also had coding style issues in my last PR so I was trying to find relevant documentations and didn't find any.

**Comment by thaigillespie** [comment]
> 
> 
> @thaigillespie just wondering if we have a document for general rules and coding styles like these so that we're more aware of the styles while we code and don't have to change it every time in PR. I also had coding style issues in my last PR so I was trying to find relevant documentations and didn't find any.

I went through the comments I left in the PR, and it looks like a lot of those things you can derive from existing code, i.e. the Page thingy. It would be way too much effort to try and enforce specific styling on every little thing. The documentation would be way too long, which would result in most people not reading it anyway. Please just practice common sense in terms of styling when you write code, use existing code and files to guide you through it and use past experiences with being corrected in terms of styling.

No, I'm not perfect, I'm not gonna catch everything, nor will I try to. I will try to only mention things that really catch my eyes, like the space between non-function const assignment thing. Plus, if we really want a standardized coding style guide, I want everyone to contribute and participate in the process of building it, not just me pumping these docs out. Currently we don't have time for that with our schedule, so I don't think we should prioritize this.

As for the imports being out of order, not sure why this isn't automatically done by ESLint bc I put the rules in for it already. Guessing something to do with ppl (including me) not running `npm i` when we start on a new branch.

I can write docs on what tools/hooks/components to use when doing certain things, though that will be a bit bc I have other things to do right now.

**Comment by audreygao** [comment]
> > @thaigillespie just wondering if we have a document for general rules and coding styles like these so that we're more aware of the styles while we code and don't have to change it every time in PR. I also had coding style issues in my last PR so I was trying to find relevant documentations and didn't find any.
> 
> I went through the comments I left in the PR, and it looks like a lot of those things you can derive from existing code, i.e. the Page thingy. It would be way too much effort to try and enforce specific styling on every little thing. The documentation would be way too long, which would result in most people not reading it anyway. Please just practice common sense in terms of styling when you write code, use existing code and files to guide you through it and use past experiences with being corrected in terms of styling.
> 
> No, I'm not perfect, I'm not gonna catch everything, nor will I try to. I will try to only mention things that really catch my eyes, like the space between non-function const assignment thing. Plus, if we really want a standardized coding style guide, I want everyone to contribute and participate in the process of building it, not just me pumping these docs out. Currently we don't have time for that with our schedule, so I don't think we should prioritize this.
> 
> As for the imports being out of order, not sure why this isn't automatically done by ESLint bc I put the rules in for it already. Guessing something to do with ppl (including me) not running `npm i` when we start on a new branch.
> 
> I can write docs on what tools/hooks/components to use when doing certain things, though that will be a bit bc I have other things to do right now.

Oh I was just wondering if we already have one and worried that I missed out something. I agree that to write a new one would be way too much work too.

**Comment by thaigillespie** [comment]
> 
> 
> > > @thaigillespie just wondering if we have a document for general rules and coding styles like these so that we're more aware of the styles while we code and don't have to change it every time in PR. I also had coding style issues in my last PR so I was trying to find relevant documentations and didn't find any.
> > 
> > 
> > I went through the comments I left in the PR, and it looks like a lot of those things you can derive from existing code, i.e. the Page thingy. It would be way too much effort to try and enforce specific styling on every little thing. The documentation would be way too long, which would result in most people not reading it anyway. Please just practice common sense in terms of styling when you write code, use existing code and files to guide you through it and use past experiences with being corrected in terms of styling.
> > No, I'm not perfect, I'm not gonna catch everything, nor will I try to. I will try to only mention things that really catch my eyes, like the space between non-function const assignment thing. Plus, if we really want a standardized coding style guide, I want everyone to contribute and participate in the process of building it, not just me pumping these docs out. Currently we don't have time for that with our schedule, so I don't think we should prioritize this.
> > As for the imports being out of order, not sure why this isn't automatically done by ESLint bc I put the rules in for it already. Guessing something to do with ppl (including me) not running `npm i` when we start on a new branch.
> > I can write docs on what tools/hooks/components to use when doing certain things, though that will be a bit bc I have other things to do right now.
> 
> Oh I was just wondering if we already have one and worried that I missed out something. I agree that to write a new one would be way too much work too.

Oh yeah we don't have one

**Review by thaigillespie — APPROVED** [review]
lgtm

**Pull request merged**